### ISSUE TITLE
DefaultControllerBuilderTest is racy

### ISSUE THREAD

**Issue by brendandburns** [kubernetes-client/java#2010]
```
Error:  Failures: 
Error:    DefaultControllerBuilderTest.testBuildWatchEventNotificationShouldWork:210 expected:<1> but was:<0>
```

See #2007

**Comment by brendandburns** [comment]
Closed by #2012

**Comment by brendandburns** [comment]
Flaked again:
```
Error:  Tests run: 5, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 3.594 s <<< FAILURE! -- in io.kubernetes.client.extended.controller.builder.DefaultControllerBuilderTest
Error:  io.kubernetes.client.extended.controller.builder.DefaultControllerBuilderTest.testWithLeaderElectorProxiesDefaultController -- Time elapsed: 0.093 s <<< FAILURE!
com.github.tomakehurst.wiremock.client.VerificationException: 
A request was unmatched by any stub mapping. Request was: {
  "url" : "/api/v1/pods?resourceVersion=0&timeoutSeconds=332&watch=true",
  "absoluteUrl" : "http://localhost:8089/api/v1/pods?resourceVersion=0&timeoutSeconds=332&watch=true",
  "method" : "GET",
  "clientIp" : "0:0:0:0:0:0:0:1",
  "headers" : {
    "User-Agent" : "Kubernetes Java Client/20.0.0-SNAPSHOT",
    "Connection" : "keep-alive",
    "Host" : "localhost:8089",
    "Accept-Encoding" : "gzip",
    "Accept" : "application/json"
  },
  "cookies" : { },
  "browserProxyRequest" : false,
  "loggedDate" : 1706841917030,
  "bodyAsBase64" : "",
  "body" : "",
  "scheme" : "http",
  "host" : "localhost",
  "port" : 8089,
  "loggedDateString" : "2024-02-02T02:45:17Z",
  "queryParams" : {
    "resourceVersion" : {
      "key" : "resourceVersion",
      "values" : [ "0" ]
    },
    "timeoutSeconds" : {
      "key" : "timeoutSeconds",
      "values" : [ "332" ]
    },
    "watch" : {
      "key" : "watch",
      "values" : [ "true" ]
    }
  }
}
	at com.github.tomakehurst.wiremock.client.VerificationException.forUnmatchedRequests(VerificationException.java:92)
	at com.github.tomakehurst.wiremock.junit.WireMockRule.checkForUnmatchedRequests(WireMockRule.java:96)
	at com.github.tomakehurst.wiremock.junit.WireMockRule.access$100(WireMockRule.java:34)
	at com.github.tomakehurst.wiremock.junit.WireMockRule$1.evaluate(WireMockRule.java:80)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:345)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```
